packages/grafana-ui/src/components/IconButton/IconButton.tsx

```diff
@@ -15,76 +15,92 @@ export type IconButtonVariant = 'primary' | 'secondary' | 'destructive';
 
 type LimitedIconSize = ComponentSize | 'xl';
 
-export interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+interface BaseProps extends Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'aria-label'> {
   /** Name of the icon **/
   name: IconName;
   /** Icon size - sizes xxl and xxxl are deprecated and when used being decreased to xl*/
   size?: IconSize;
   /** Type of the icon - mono or default */
   iconType?: IconType;
-  /** Tooltip content to display on hover */
-  tooltip?: PopoverContent;
-  /** Position of the tooltip */
-  tooltipPlacement?: TooltipPlacement;
   /** Variant to change the color of the Icon */
   variant?: IconButtonVariant;
-  /** Text available only for screen readers. Will use tooltip text as fallback. */
+}
+
+interface BasePropsWithTooltip extends BaseProps {
+  /** Tooltip content to display on hover and as the aria-label */
+  tooltip: PopoverContent;
+  /** Position of the tooltip */
+  tooltipPlacement?: TooltipPlacement;
+}
+
+interface BasePropsWithAriaLabel extends BaseProps {
+  /** @deprecated use aria-label instead*/
   ariaLabel?: string;
+  /** Text available only for screen readers. No tooltip will be set in this case. */
+  ['aria-label']: string;
 }
 
-export const IconButton = React.forwardRef<HTMLButtonElement, Props>(
-  (
-    {
-      name,
-      size = 'md',
-      iconType,
-      tooltip,
-      tooltipPlacement,
-      ariaLabel,
-      className,
-      variant = 'secondary',
-      ...restProps
-    },
-    ref
-  ) => {
-    const theme = useTheme2();
-    let limitedIconSize: LimitedIconSize;
-
-    // very large icons (xl to xxxl) are unified to size xl
-    if (size === 'xxl' || size === 'xxxl') {
-      deprecationWarning('IconButton', 'size="xxl" and size="xxxl"', 'size="xl"');
-      limitedIconSize = 'xl';
-    } else {
-      limitedIconSize = size;
-    }
-
-    const styles = getStyles(theme, limitedIconSize, variant);
-    const tooltipString = typeof tooltip === 'string' ? tooltip : '';
-
-    // When using tooltip, ref is forwarded to Tooltip component instead for https://github.com/grafana/grafana/issues/65632
-    const button = (
+export type Props = BasePropsWithTooltip | BasePropsWithAriaLabel;
```

Comment:
I like this. It really explains the `if ('tooltip' in props)` part further down.